### PR TITLE
Added support for rebuild_config in hostgroup

### PR DESCRIPTION
**Pull request by vijay8451**
- Added support for rebuild_config in hostgroup
- test result:
```
(Pdb) dir(hostgroup)
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_fields', '_meta', '_server_config', 'add_puppetclass', 'architecture', 'clone', 'compare', 'content_source', 'content_view', 'create', 'create_json', 'create_missing', 'create_payload', 'create_raw', 'delete', 'delete_puppetclass', 'delete_raw', 'domain', 'environment', 'get_fields', 'get_values', 'id', 'lifecycle_environment', 'list_scparams', 'list_smart_variables', 'location', 'medium', 'name', 'operatingsystem', 'organization', 'parent', 'path', 'ptable', 'puppet_ca_proxy', 'puppet_proxy', 'read', 'read_json', 'read_raw', 'realm', 'rebuild_config', 'search', 'search_filter', 'search_json', 'search_normalize', 'search_payload', 'search_raw', 'subnet', 'to_json', 'to_json_dict', 'update', 'update_json', 'update_payload', 'update_raw']
(Pdb) 

(Pdb) hostgroup.rebuild_config(data={'only': 'DNS'})
{'message': 'Configuration successfully rebuilt.'}
(Pdb) hostgroup.rebuild_config(data={'only': 'TFTP'})
{'message': 'Configuration successfully rebuilt.'}
(Pdb) 
```

### CODE REVIEW

**Comment by oshtaier** [comment]
is it possible to read that configuration value? I mean to see that it was really changed?

**Comment by jyejare** [comment]
@vijay8451, Check for all the possible unit tests. I think you may need to satisfy more tests in unit tests.

**Comment by vijay8451** [comment]
> is it possible to read that configuration value? I mean to see that it was really changed?
@oshtaier  I checked on that part but it does not provide in read:
```
(Pdb) re_config = hostgroup.rebuild_config(data={'only': 'TFTP'})
(Pdb) re_config
{'message': 'Configuration successfully rebuilt.'}

(Pdb) hostgroup.read()
nailgun.entities.HostGroup(architecture=None, domain=None, puppet_proxy=None, puppet_ca_proxy=None, content_source=None, environment=None, lifecycle_environment=None, location=[nailgun.entities.Location(id=60)], medium=None, name='MRfyIsqxjva', operatingsystem=None, organization=[nailgun.entities.Organization(id=59)], parent=None, ptable=None, realm=None, subnet=None, content_view=None, id=9)
(Pdb) 
```

**Comment by vijay8451** [comment]
> @vijay8451, Check for all the possible unit tests. I think you may need to satisfy more tests in unit tests.

@jyejare I checked and found the test under ``

> @vijay8451, Check for all the possible unit tests. I think you may need to satisfy more tests in unit tests.


@jyejare I checked on that but did not find anything much around it.

**Comment by jyejare** [comment]
@vijay8451 , You may add here as well : https://github.com/SatelliteQE/nailgun/blob/063316a46dcfc32022cc0775ab949986f3843605/tests/test_entities.py#L402

**Comment by vijay8451** [comment]
@jyejare thanks , added in recent commit.

**Comment by vijay8451** [comment]
@jyejare added a test test_rebuild_config, please review.

**Comment by jyejare** [comment]
> @jyejare added a test test_rebuild_config, please review.

Fantastic work! :+1:  Thanks for addressing valuable tests and comments.

File: tests/test_entities.py [hunk 1]
```diff
@@ -3182,6 +3184,29 @@ def test_clone_hostgroup(self):
         self.assertEqual(handler.call_count, 1)
         self.assertEqual(handler.return_value, response)
 
+    def test_rebuild_config(self):
+        """"Test for :meth:`nailgun.entities.HostGroup.rebuild_config`
+                Assert that the method is called one with correct argumets
```
[the remark this answers is not in the document]
Comment (vijay8451) [reply tests/test_entities.py]: Addressed in latest commit.

**Comment by ntkathole** [comment]
@vijay8451 Thanks. Please cherry-pick for stable.

**Pull request merged**